### PR TITLE
Refs #37130 - Set the mode on the generated bootstrap RPM

### PR DESCRIPTION
**Pull request by ehelms**
@evgeni I should have tried harder to test it the first time.

### CODE REVIEW

**Comment by ehelms** [comment]
We cannot use an implicit file resource here because the file that is generated depends on build time results of the rpmbuild process.

File: lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb [hunk 1]
```diff
@@ -100,6 +100,8 @@ def build_rpm
 
   def copy_rpm
     FileUtils.copy(built_rpm, resource[:dest])
+    File.chmod(resource[:mode].to_i(8), latest_rpm)
+    FileUtils.chown(resource[:owner], resource[:group], latest_rpm)
```
Comment (wbclark) [inline lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb:104]:
```suggestion
    latest = latest_rpm
    File.chmod(resource[:mode].to_i(8), latest)
    FileUtils.chown(resource[:owner], resource[:group], latest)
```

avoids repeating some (relatively) expensive operations in latest_rpm

**Comment by evgeni** [comment]
Should we do the same for the SRPM for consistency? It's not needed during deployment, but we do publish it in pub and if it's not readable it's useless.

File: lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb [hunk 2]
```diff
@@ -100,12 +102,19 @@ def build_rpm
...
   end
 
   def copy_srpm
     FileUtils.copy(built_srpm, resource[:dest])
   end
 
+  def set_ownership
+    File.chmod(resource[:mode].to_i(8), latest)
+    FileUtils.chown(resource[:owner], resource[:group], latest)
+  end
```
Comment (wbclark) [inline lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb:116]:
```suggestion
  def set_ownership(rpm_filepath)
    File.chmod(resource[:mode].to_i(8), rpm_filepath)
    FileUtils.chown(resource[:owner], resource[:group], rpm_filepath)
  end
```

File: lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb [hunk 3]
```diff
@@ -100,12 +102,19 @@ def build_rpm
 
   def copy_rpm
     FileUtils.copy(built_rpm, resource[:dest])
+    set_ownership(latest_rpm)
+    set_ownership(latest_rpm(source: true))
   end
```
Comment (wbclark) [inline lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb:107]:
```suggestion
  def copy_rpm
    FileUtils.copy(built_rpm, resource[:dest])
    set_ownership(latest_rpm)
  end
```

File: lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb [hunk 4]
```diff
@@ -100,12 +102,19 @@ def build_rpm
...
 
   def copy_srpm
     FileUtils.copy(built_srpm, resource[:dest])
   end
```
Comment (wbclark) [inline lib/puppet/provider/bootstrap_rpm/bootstrap_rpm.rb:111]:
```suggestion
  def copy_srpm
    FileUtils.copy(built_srpm, resource[:dest])
    set_ownership(latest_rpm(source: true)
  end
```

**Review by wbclark — COMMENTED** [review]
In for a dime, in for a dollar...

**Review by evgeni — APPROVED** [review]
Nice!

**Review by wbclark — APPROVED** [review]
LGTM

**Pull request merged**